File: C++/Numerics/NumericsBenchmark.cpp
#include "AbstractBenchmark.cpp"

namespace oobench {

    class NumericsBenchmark : public AbstractBenchmark {
    public:
        NumericsBenchmark(int argc, const char** argv) 
                : AbstractBenchmark(argc, argv) { 
        }

        virtual int getMajorNumber() const {
            return 15;
        }

        virtual void algorithm(unsigned long count) = 0;

        void test(unsigned long items, const char* description) {
            unsigned long count = getCountWithDefault(items);
            std::cout << "*** Testing " << description << std::endl;
            algorithm(count);
        }
    };
}

File: Java/org/oobench/numerics/NumericsBenchmark.java
package org.oobench.numerics;

import org.oobench.common.*;

public abstract class NumericsBenchmark extends AbstractBenchmark {

    public int getMajorNumber() {
        return 15;
    }

    public abstract void algorithm(int count);

    public void test(int count, String description) {
        count = getCountWithDefault(count);
        System.out.println("*** Testing " + description);

        algorithm(count);
    }

    public static void testNumerics(Class theClass, int count, 
            String description, String[] args) {
        try {
            NumericsBenchmark bench = 
                    (NumericsBenchmark)theClass.newInstance();
            bench.setArguments(args);
            bench.test(count, description);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}

